src/Lib/package.cpp

```diff
@@ -34,17 +40,61 @@ const std::string Package::GetFullyQualifiedName(void) {
   return Package::GetName() + " " + Package::GetVersion();
 }
 
+const void Package::assignDataDir(void) {
+  const char *possible[] = {".", "..", DATADIR, NULL};
+  int i = 0;
+
+  while (possible[i] != NULL) {
+    char map[1024] = "";
+    size_t r = snprintf(map, sizeof map, "%s/maps", possible[i]);
```

Comment:
Can we use std::string for string concatenation? Removes 1023 char limit here

Reply:
Done.